Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +126,87 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::startsWith($path, '/', 'Target path "%s" must be absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentDocument = $this->getManager()->find(null, $resolvedPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $resolvedPath));
+        }
+
+        /** @var PhpcrOdmResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::same($resolvedPath, $this->resolvePath($resource->getPath()));
+
+            $this->getManager()->persist($resource->getPayload());
+        }
+
+        $this->getManager()->flush();
+    }
+
+    /**
+     * Moves a resource inside the repository.
+     *
+     * @param string $sourceQuery The Path of the current document.
+     * @param string $targetPath  The parent path of the destination.
+     * @param string $language
+     *
+     * @return int
+     */
+    public function move($sourceQuery, $targetPath, $language = 'glob')
```

Comment:
yes that is exactly the reason why there is no {@inheritdoc}.

I would say :+1:  for your idea.